packages/node/src/soroban/safe-api.ts

```diff
@@ -0,0 +1,111 @@
+// Copyright 2020-2022 OnFinality Limited authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import { getLogger } from '@subql/node-core';
+import {
+  Account,
+  Address,
+  Contract,
+  FeeBumpTransaction,
+  Server,
+  SorobanRpc,
+  Transaction,
+  xdr,
+} from 'soroban-client';
+import { Durability, GetEventsRequest } from 'soroban-client/lib/server';
+import { SorobanServer } from './soroban.server';
+
+const logger = getLogger('safe.api.soroban');
+
+export default class SafeSorobanProvider extends SorobanServer {
```

Comment:
The purpose of the safe client is to query contract state at the blockHeight. If we can't do that there's no point in having it.

Reply:
the api only provides the current state of a contract

https://soroban.stellar.org/api/methods/getLedgerEntries